File: isomorphic-strings/isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        n = len(s)
        m = len(t)
        if n != m:
            return False
        mp = {}
        mp1 = {}
        for idx in range(n):
            if s[idx] in mp:
                if mp[s[idx]] != t[idx]:
                    return False
            else:
                if t[idx] in mp1 and mp1[t[idx]]:
                    return False
                mp[s[idx]] = t[idx]
                mp1[t[idx]] = True
        return True

File: isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n=s.length(),m=t.length();
        if(n!=m) return false;
        unordered_map<char,char>mp;
        unordered_map<char,bool>mp1;
        for(int idx=0;idx<n;idx++){
            if(mp.find(s[idx])!=mp.end()){
                if(mp[s[idx]]!=t[idx]) return false;
            }
            else{
                if(mp1[t[idx]]) return false;
                mp[s[idx]]=t[idx];
                mp1[t[idx]]=true;
            }
        }
        return true;
    }
};